packages/unchained-client/src/evm/parser/index.ts

```diff
@@ -135,8 +142,11 @@ export class BaseTransactionParser<T extends Tx> {
       }
 
       const assetId = (() => {
-        // alias ether token on optimism to native asset as they are the same
-        if (transfer.contract === '0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000') {
+        // alias LegacyERC20ETH on optimism to native asset as they are the same (pre bedrock)
+        if (
+          this.chainId === optimismChainId &&
+          transfer.contract === '0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000'
+        ) {
```

Comment:
Possibly separate diff too to keep things clean, though this is a super straightforward change so meh

Reply:
this is relevant, as the same address exists on base, but is not the legacy eth contract as base was created after bedrock upgrade